promptulate/tools/file/tools.py

```diff
@@ -0,0 +1,250 @@
+        except PermissionError:
+            return "Permission denied: You don't have permission to write to this file."
+        return "Write file successfully"
+
+
+class AppendFileTool(Tool):
+    name: str = "append-file"
+    description: str = (
+        "Append to a file"
+        "Useful when you need to edit/create edit a file."
+        "It can edit the specified file in the specified directory/local directory,"
+        "Append the content to the end of the file."
+        "If the file does not exist, edit it after it is created"
+        "The output parameters must be:"
+        "file_name(file name), text(related information)."
+    )
+
+    def __init__(self, root_dir: str, *args, **kwargs) -> None:
+        self.root_dir = root_dir
+        super().__init__(*args, **kwargs)
+
+    def _run(self, file_name: str, text: str) -> str:
+        """Append to a file
+
+        Args:
+            file_name(Optional(str)): The name of the file you want to edit.
+            text(Optional(str)): The content you want to edit.
+
+        Returns:
+            str: Append file successfully
+        """
+        try:
+            with open(f"{self.root_dir}/{file_name}", "a") as file:
+                file.write(text)
+        except IOError:
+            return "An error occurred while trying to write to the file."
+        except PermissionError:
+            return "Permission denied: You don't have permission to write to this file."
+        return "Append file successfully"
+
+
+class ReadFileTool(Tool):
+    name: str = "read-file"
+    description: str = (
+        "Read a file"
+        "Useful when you need to read a file."
+        "It can read the specified file in the specified directory/local directory."
+        "The output parameters must be: file_name(file name)."
+    )
+
+    def __init__(self, root_dir: str, *args, **kwargs) -> None:
+        self.root_dir = root_dir
+        super().__init__(*args, **kwargs)
+
+    def _run(self, file_name: str) -> str:
+        """Read a file
+
+        Args:
+            file_name(Optional(str)): The name of the file you want to read.
+
+        Returns:
+            str: The contents of the file.
+        """
+        try:
+            with open(f"{self.root_dir}/{file_name}", "r") as file:
+                return file.read()
+        except FileNotFoundError:
+            return "File not found: The file you are trying to read does not exist."
+        except PermissionError:
+            return "Permission denied: You don't have permission to read this file."
+        except UnicodeDecodeError:
+            return (
+                "Unicode decode error:"
+                "The file could not be decoded with utf-8 encoding."
+            )
+        except IOError:
+            return "An error occurred while trying to read the file."
+
+
+class DeleteFileTool(Tool):
+    name: str = "delete-file"
+    description: str = (
```

Comment:
All tool should add parameters, parameters description should not declare in `description`. It means you should remove `"The output parameters must be: file_name(file name)."` 

See https://github.com/Undertone0809/promptulate/blob/d0af1e59d6f67b9f35a3cab560fac9a1ccd57442/tests/tools/test_tool.py#L172 to see how to declare a tool.